src/Columns/ColumnNullable.cpp

```diff
@@ -134,21 +136,34 @@ void ColumnNullable::insertData(const char * pos, size_t length)
     }
 }
 
-StringRef ColumnNullable::serializeValueIntoArena(size_t n, Arena & arena, char const *& begin) const
+StringRef ColumnNullable::serializeValueIntoArena(size_t n, Arena & arena, char const *& begin, const UInt8 *) const
 {
     const auto & arr = getNullMapData();
     static constexpr auto s = sizeof(arr[0]);
+    char * pos;
+    if (is_string)
+    {
+        const auto * column_string = static_cast<const ColumnString *>(nested_column.get());
+        return column_string->serializeValueIntoArena(n, arena, begin, &arr[n]);
+    }
+    else if (is_fixed_size_column)
```

Comment:
isn't it (almost) always a `ColumnVector`? then we could do cast here also and avoid virt call.

Reply:
ColumnVector is a generic type. Specifically, there are 14 types in DatatypesNumber.h. If expanded in sequence, it may not be much different from calling a virtual function. At the same time, Decimal and FixedString are also supported here.